crates/shared/src/trade_finding/oneinch.rs

```diff
@@ -24,6 +28,8 @@ struct Inner {
     disabled_protocols: Vec<String>,
     protocol_cache: ProtocolCache,
     referrer_address: Option<H160>,
+    spender_cache: Arc<Mutex<Option<(H160, Instant)>>>,
```

Comment:
Ahh, yes. That's true. I thought if I don't use an `Option` here I would have to properly initiate the `spender_cache` in the constructor which would mean it can fail. But using the `Default` initialization is actually quite nice!